File: cloudflare-worker/src/predictors/strategies/qxc/balanced.js
/**
 * 七星彩均衡策略
 */

import { BaseStrategy } from './base.js';

export class BalancedStrategy extends BaseStrategy {
  constructor() {
    super('balanced', '均衡策略', '追求号码分布均衡，大小号、奇偶号均衡');
  }

  generateNumbers(context) {
    const numbers = [];
    
    // 确保大小号均衡（0-4为小，5-9为大）
    // 确保奇偶号均衡
    const smallNumbers = [0, 1, 2, 3, 4];
    const largeNumbers = [5, 6, 7, 8, 9];
    const oddNumbers = [1, 3, 5, 7, 9];
    const evenNumbers = [0, 2, 4, 6, 8];
    
    // 3-4个小号，3-4个大号
    const smallCount = 3 + Math.floor(Math.random() * 2);
    const largeCount = 7 - smallCount;
    
    // 生成小号
    for (let i = 0; i < smallCount; i++) {
      numbers.push(this.randomChoice(smallNumbers));
    }
    
    // 生成大号
    for (let i = 0; i < largeCount; i++) {
      numbers.push(this.randomChoice(largeNumbers));
    }
    
    // 打乱顺序
    return this.shuffle(numbers);
  }

  shuffle(array) {
    const result = [...array];
    for (let i = result.length - 1; i > 0; i--) {
      const j = Math.floor(Math.random() * (i + 1));
      [result[i], result[j]] = [result[j], result[i]];
    }
    return result;
  }
}

File: lotteries/qxc/strategies/balanced.py
"""
均衡策略 - 大小号均衡
"""

from .base import BaseStrategy
from typing import List, Dict
import random


class BalancedStrategy(BaseStrategy):
    """均衡策略：大小号均衡分布"""
    
    def __init__(self):
        super().__init__(
            name='均衡策略',
            description='大小号均衡分布（0-4为小号，5-9为大号）'
        )
    
    def generate_numbers(self, context: Dict) -> List[int]:
        """生成7个号码"""
        small_numbers = list(range(0, 5))  # 0-4
        large_numbers = list(range(5, 10))  # 5-9
        
        numbers = []
        
        # 3-4个小号，3-4个大号
        small_count = random.choice([3, 4])
        large_count = 7 - small_count
        
        for _ in range(small_count):
            numbers.append(random.choice(small_numbers))
        
        for _ in range(large_count):
            numbers.append(random.choice(large_numbers))
        
        # 打乱顺序
        random.shuffle(numbers)
        
        return numbers
